run-test.rb

```diff
@@ -1,5 +1,6 @@
 #!/usr/bin/env ruby
 
+require "bundler/setup"
```

Comment:
If we add this, we always use local strscan not strscan gem.

https://github.com/ruby/strscan/actions/runs/4160453753/jobs/7197443977#step:11:5

```text
Loaded strscan from /home/runner/work/strscan/strscan/lib/strscan.so
Gem from /home/runner/work/strscan/strscan
```

We should use strscan gem like https://github.com/ruby/strscan/actions/runs/4134931150/jobs/7146703286#step:11:5 :

```text
Loaded strscan from /opt/hostedtoolcache/Ruby/3.1.3/x64/lib/ruby/gems/3.1.0/gems/strscan-3.0.7/lib/strscan.so
Gem from /opt/hostedtoolcache/Ruby/3.1.3/x64/lib/ruby/gems/3.1.0/gems/strscan-3.0.7
```

Reply:
I see, I reverted that change with force push.
Should I do something to make CI pass? I believe after releasing a new gem including this fix Ci will become green.
Or more general question, what should we do if a fix and a test are included in the same commit?

Reply:
> Should I do something to make CI pass?

Can you take a look at why installed gem isn't used only with Ruby 3.2 and master?
Or could you create an issue for this?

It may be a RubyGems/Bundler bug in Ruby 3.2 and master... 

> I believe after releasing a new gem including this fix Ci will become green.

It's not the intention for the test. The test should use "not-released" gem to prevent releasing a buggy gem.

>  Or more general question, what should we do if a fix and a test are included in the same commit?

If our CI jobs work as expected (installed gem is used), our test will be passed in the case.

Reply:
OK so this might be an issue outside of strscan, now I understand.
I'll try to resolve it myself, and if it's too difficult I'll submit an issue.